bin/grid

```diff
@@ -98,6 +96,8 @@ start_all() {
   $DIR/grid start zookeeper
   #$DIR/grid start yarn
   $DIR/grid start kafka
+  #We need to add a sleep to allow kafka to start before starting schemaregistry otherwise a connect exception is thrown
+  sleep 3
```

Comment:
How about waiting for the socket to be listening?

wait_kafka() {
  echo "Waiting for Kafka broker..."
  until $(nc -w 2 localhost 9092); do
    printf '.'
    sleep 1
  done
  echo "Kafka broker is listening\n";
}

start_schemaregistry() {
  if [ -f $DEPLOY_ROOT_DIR/confluent/bin/schema-registry-start ]; then
    mkdir -p $DEPLOY_ROOT_DIR/$SYSTEM/logs
    pushd $DEPLOY_ROOT_DIR/$SYSTEM/logs
      wait_kafka

On Wed, Apr 13, 2016 at 10:46 AM, Vishal Kuo <notifications@github.com>
wrote:

> In bin/grid
> <https://github.com/theduderog/hello-samza-confluent/pull/10#discussion_r59592861>
> :
>
> > @@ -98,6 +96,8 @@ start_all() {
> >    $DIR/grid start zookeeper
> >    #$DIR/grid start yarn
> >    $DIR/grid start kafka
> > +  #We need to add a sleep to allow kafka to start before starting schemaregistry otherwise a connect exception is thrown
> > +  sleep 3
>
> If there is a better alternative to this solution please let me know
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/theduderog/hello-samza-confluent/pull/10/files/286b73c7891a806a9aef59821188381c43344e6f#r59592861>
>